packages/block-library/src/navigation-link/edit.js

```diff
@@ -111,14 +111,14 @@ const useIsInvalidLink = ( kind, type, id ) => {
 			// Fetching the posts status is an "expensive" operation. Especially for sites with large navigations.
 			// When the block is rendered in a template or other disabled contexts we can skip this check in order
 			// to avoid all these additional requests that don't really add any value in that mode.
-			if ( blockEditingMode === 'disabled' ) {
+			if ( blockEditingMode === 'disabled' || ! enabled ) {
```

Comment:
Left `blockEditingMode` check. This handles cases when the Navigation block can be selected, but its children have editing disabled.